### PR TITLE
Make null kafka config errors explicit

### LINKED ISSUE #1236: make configuration errors on kafka.* properties more explicit
Currently if there is any kafka.<property> configured (mistakingly, cause yaml) as null, then it would be helpful to specify which one is it, right now it just says NullPointerException.

This becomes more problematic if there is any user added property like kafka.custom.users.

### PR DESCRIPTION
**Pull request by graemerocher**
## Summary

- fail fast when a kafka.* property resolves to null
- report the full kafka property path in the configuration error
- add a regression test covering a custom null kafka property
 
Resolves #1236

### CODE REVIEW

**Comment by graemerocher** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/micronaut-projects/micronaut-kafka/pull/1318#pullrequestreview-4146165999)

**Comment by graemerocher** [comment]
@copilot resolve the merge conflicts in this pull request

**Comment by graemerocher** [comment]
@copilot resolve the merge conflicts in this pull request

File: kafka/src/main/java/io/micronaut/configuration/kafka/config/AbstractKafkaConfiguration.java [hunk 1]
```diff
@@ -82,8 +83,11 @@ protected static Properties toKafkaProperties(Environment environment, Map<?, ?>
             return Stream.of("embedded", "consumers", "producers", "streams").noneMatch(key::startsWith);
         }).forEach(entry -> {
             Object value = entry.getValue();
-            if (environment.getConversionService().canConvert(entry.getValue().getClass(), String.class)) {
-                Optional<?> converted = environment.getConversionService().convert(entry.getValue(), String.class);
+            if (value == null) {
+                throw new ConfigurationException("Value for property " + PREFIX + "." + entry.getKey().toString() + " resolved as null");
```
Comment (radovanradic) [inline kafka/src/main/java/io/micronaut/configuration/kafka/config/AbstractKafkaConfiguration.java:87]: If there is `kafka.streams.*` null properties would it report wrong prefix and key?

Comment (graemerocher) [reply kafka/src/main/java/io/micronaut/configuration/kafka/config/AbstractKafkaConfiguration.java:87]: I don't believe so, the change is minimal and just ensures null isn't past. see the reported issue

**Review by radovanradic — APPROVED** [review]

**Comment by radovanradic** [comment]
The build is failing but probably not related to the changes

**Comment by graemerocher** [comment]
I fixed CI in https://github.com/micronaut-projects/micronaut-kafka/commit/b028537c080250c1615c7af132c792ca8a14eb28
